internal/metrics/middleware.go

```diff
@@ -79,10 +116,37 @@ func HTTPMetricsHandler(service string) func(next http.Handler) http.Handler {
 				tag.Insert(keyStatus, strconv.Itoa(m.Status())),
 			)
 			stats.Record(ctx,
-				httpRequestCount.M(1),
-				httpRequestDuration.M(time.Since(startTime).Nanoseconds()/int64(time.Millisecond)),
-				httpResponseSize.M(int64(m.BytesWritten())),
+				httpServerRequestCount.M(1),
+				httpServerRequestDuration.M(time.Since(startTime).Nanoseconds()/int64(time.Millisecond)),
+				httpServerResponseSize.M(int64(m.BytesWritten())),
+			)
+		})
+	}
+}
+
+// HTTPMetricsRoundTripper creates a metrics tracking tripper for outbound HTTP Requests
+func HTTPMetricsRoundTripper(service string) func(next http.RoundTripper) http.RoundTripper {
+
+	return func(next http.RoundTripper) http.RoundTripper {
+		return tripper.RoundTripperFunc(func(r *http.Request) (*http.Response, error) {
+			startTime := time.Now()
+
+			resp, err := next.RoundTrip(r)
```

Comment:
If an error gets thrown here, I assume `resp` would be nil and the next two lines may panic due to nil reference?

Reply:
That's a good point.  I was thinking on the server side still...and that a response of some sort would happen - but that's not true here in the Transport layer.  Will fix.